diff -up a/meson.build b/meson.build
--- a/meson.build	2021-03-31 04:59:02.000000000 +0800
+++ b/meson.build	2021-12-10 10:59:07.586868800 +0800
@@ -3076,6 +3076,7 @@
         if want_tests != 'false'
                 test('test-sysusers',
                      test_sysusers_sh,
+                     timeout : 120,
                      # https://github.com/mesonbuild/meson/issues/2681
                      args : exe.full_path())
         endif
@@ -3322,7 +3323,7 @@
         type = tuple.length() > 5 ? tuple[5] : ''
         defs = tuple.length() > 6 ? tuple[6] : []
         parallel = tuple.length() > 7 ? tuple[7] : true
-        timeout = 30
+        timeout = 180
 
         name = sources[0].split('/')[-1].split('.')[0]
         if type.startswith('timeout=')
