From 6afc31615e63b7db941684be84da82a06373a778 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 9 Jan 2020 17:01:46 +0100
Subject: [PATCH] core: no need to initialize swap structure fields if all
 zeroes anyway

---
 src/core/swap.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/core/swap.c b/src/core/swap.c
index 03f443daec..6caf20ea66 100644
--- a/src/core/swap.c
+++ b/src/core/swap.c
@@ -63,7 +63,6 @@ static void swap_unset_proc_swaps(Swap *s) {
                 return;
 
         s->parameters_proc_swaps.what = mfree(s->parameters_proc_swaps.what);
-
         s->from_proc_swaps = false;
 }
 
@@ -117,9 +116,6 @@ static void swap_init(Unit *u) {
         s->exec_context.std_output = u->manager->default_std_output;
         s->exec_context.std_error = u->manager->default_std_error;
 
-        s->parameters_proc_swaps.priority = s->parameters_fragment.priority = 0;
-        s->parameters_fragment.priority_set = false;
-
         s->control_command_id = _SWAP_EXEC_COMMAND_INVALID;
 
         u->ignore_on_isolate = true;
-- 
2.23.0

