From 7424b6c0f38d4a32fd96e74d7078707c026c6c66 Mon Sep 17 00:00:00 2001
From: wangyuhang <wangyuhang27@huawei.com>
Date: Thu, 9 Jun 2022 20:10:50 +0800
Subject: [PATCH] keep weight consistent with the set value

---
 src/core/cgroup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/cgroup.c b/src/core/cgroup.c
index ee15d50..4c5feef 100644
--- a/src/core/cgroup.c
+++ b/src/core/cgroup.c
@@ -1249,7 +1249,7 @@ static void set_io_weight(Unit *u, uint64_t weight) {
          * See also: https://github.com/systemd/systemd/pull/13335 and
          * https://github.com/torvalds/linux/commit/65752aef0a407e1ef17ec78a7fc31ba4e0b360f9.
          * The range is 1..1000 apparently, and the default is 100. */
-        xsprintf(buf, "%" PRIu64 "\n", BFQ_WEIGHT(weight));
+        xsprintf(buf, "%" PRIu64 "\n", weight);
         (void) set_attribute_and_warn(u, "io", "io.bfq.weight", buf);
 
         xsprintf(buf, "default %" PRIu64 "\n", weight);
@@ -1262,7 +1262,7 @@ static void set_blkio_weight(Unit *u, uint64_t weight) {
         assert(u);
 
         /* FIXME: see comment in set_io_weight(). */
-        xsprintf(buf, "%" PRIu64 "\n", BFQ_WEIGHT(weight));
+        xsprintf(buf, "%" PRIu64 "\n", weight);
         (void) set_attribute_and_warn(u, "blkio", "blkio.bfq.weight", buf);
 
         xsprintf(buf, "%" PRIu64 "\n", weight);
-- 
2.33.0

